python/ray/tests/test_object_manager.py

```diff
@@ -375,6 +375,39 @@ def allocate(i):
     del allocated
 
 
+@pytest.mark.timeout(30)
+def test_max_pinned_args_memory(shutdown_only):
+    cluster = Cluster()
+    cluster.add_node(
+        num_cpus=0,
+        object_store_memory=200 * 1024 * 1024,
+        _system_config={
+            "max_task_args_memory_fraction": 0.7,
+        })
+    ray.init(address=cluster.address)
+    cluster.add_node(num_cpus=3, object_store_memory=100 * 1024 * 1024)
+
+    @ray.remote
+    def f(arg):
+        time.sleep(1)
+        return np.zeros(30 * 1024 * 1024, dtype=np.uint8)
+
+    # Each task arg takes about 30% of the remote node's memory. We should
+    # execute at most 2 at a time to make sure we have room for at least 1 task
+    # output.
+    x = np.zeros(30 * 1024 * 1024, dtype=np.uint8)
+    ray.get([f.remote(ray.put(x)) for _ in range(3)])
```

Comment:
This test doesn't really verify that 3 tasks are not running at the same time right?

Why don't we add `Wait for condition used_cpus == 2` before we call ray.get here?

Reply:
It does; before this test would error with ObjectStoreFullError.